components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -823,16 +824,22 @@ public static void checkPasswordPatternViolation(UserStoreException exception, U
     private static RealmConfiguration getRealmConfiguration(User user) throws IdentityRecoveryClientException {
 
         int tenantId = IdentityTenantUtil.getTenantId(user.getTenantDomain());
-        UserStoreManager userStoreManager;
         try {
-            userStoreManager = IdentityRecoveryServiceDataHolder.getInstance().getRealmService().
+            UserStoreManager userStoreManager = IdentityRecoveryServiceDataHolder.getInstance().getRealmService().
                     getTenantUserRealm(tenantId).getUserStoreManager();
+            userStoreManager = ((org.wso2.carbon.user.core.UserStoreManager) userStoreManager).getSecondaryUserStoreManager(
```

Comment:
Let's use a different variable name like secondaryUserStoreManager.

Reply:
This option was discussed, and we concluded not to use a variable name like secondaryUserStoreManager because, in other use cases, it could cause confusion. It was mentioned that the value returned here is not necessarily a secondary user store.